vassal-app/src/main/java/VASSAL/build/module/gamepieceimage/GamePieceImage.java

```diff
@@ -575,45 +575,34 @@ private boolean isPng(FilterBypass fb) {
     }
 
     /** Ensure the current value ends in '.png' */
-    private void fixPng(FilterBypass fb) throws BadLocationException {
-      fixPng(fb, 1);
-    }
-
-    private void fixPng(FilterBypass fb, int pos) throws BadLocationException {
+    private void fixPng(FilterBypass fb, int caretPos) throws BadLocationException {
       if (!isPng(fb)) {
         final String text = getText(fb);
         if (text.endsWith(".")) {
           super.replace(fb, text.length() - 1, 1, PNG_SUFFIX, null);
-          config.setCaretPosition(pos);
+          config.setCaretPosition(caretPos);
         }
         else if (text.toUpperCase().endsWith(".p")) {
```

Comment:
Am I missing something here, or is this test always going to fail (converting to uppercase, then checking for a lowercase letter)?  (Same for the next "else if", too.)